packages/jest-haste-map/src/crawlers/node.ts

```diff
@@ -84,9 +85,32 @@ function find(
 
           // This logic is unnecessary for node > v10.10, but leaving it in
           // since we need it for backwards-compatibility still.
-          if (!err && stat && !stat.isSymbolicLink()) {
+          if (!err && stat) {
             if (stat.isDirectory()) {
               search(file);
+            }
+            if (stat.isSymbolicLink()) {
```

Comment:
This pyramid code is extremely hard to follow. Any chance of refactoring and flattening it a bit?

Reply:
yep! turned out i was just being a bit thick :p